packages/ic-error-types/src/lib.rs

```diff
@@ -338,7 +338,11 @@ impl UserError {
     /// Panics if the error doesn't have the expected code and description.
     /// Useful for tests to avoid matching exact error messages.
     pub fn assert_contains(&self, code: ErrorCode, description: &str) {
-        assert_eq!(self.code, code);
+        assert_eq!(
+            self.code, code,
+            "Failed to match actual error \"{:?}\" with expected \"{}, {}\"",
+            self, code, description
+        );
```

Comment:
Unrelated change, just because I noticed it's hard to tell what's going wrong when this assert fails since we don't see the full error.

Reply:
Should we pull it in a separate PR if it's unrelated? I'd have thought that you hit it while working on this change which would then make it relevant imo.

Reply:
I did encounter it while making this change. I can put it in a separate PR though.

Reply:
Moved to https://github.com/dfinity/ic/pull/5090